### PR TITLE
Refactoring of PythonCANSockets

### PR DESCRIPTION
**Pull request by polybassa**
@gpotter2 As discussed in #3293 

This PR refactors PythonCANSockets to use ObjectPipes internally and removes a bunch of threads from the unit tests

### CODE REVIEW

**Comment by gpotter2** [comment]
That was fast ! Could you rebase against master (all of your PRs actually, tests should now be fixed...)
Note this included the disabling of tests

**Comment by polybassa** [comment]
Hi, some are rebased and pass. Could you please merge them?
-- 
Viele Grüße

Nils Weiß

File: scapy/contrib/cansocket_python_can.py [hunk 1]
```diff
@@ -115,8 +73,8 @@ def __init__(self):
         self.pool = dict()  # type: Dict[str, SocketMapper]
         self.pool_mutex = threading.Lock()
 
-    def internal_send(self, sender, msg, prio=0):
-        # type: (SocketWrapper, can_Message, int) -> None
+    def internal_send(self, sender, msg):
```
Comment (guedou) [inline scapy/contrib/cansocket_python_can.py:76]: _internal_send() might be a better name.

**Review by guedou — CHANGES_REQUESTED** [review]
Thanks for this PR! Does it improve tests stability?

**Comment by polybassa** [comment]
> Thanks for this PR! Does it improve tests stability?

I hope so. It's hard to say, because the instability is really difficult to debug. 

At least, this PR removes a lot of complexity and makes use of `ObjectPipe`, which I have the feeling might be more stable. 

After #3309 gets merged, I plan to refactor some more automotive related unit tests and remove further complexity.

File: scapy/contrib/cansocket_python_can.py (hunk 1 above)
Comment (polybassa) [reply scapy/contrib/cansocket_python_can.py:76]: Thanks

**Comment by polybassa** [comment]
Unfortunately, I could not use ObjectPipe, since the performance decreases, compared to the current solution.

**Comment by polybassa** [comment]
So, this PR still removes a lot of complexity form unit tests

**Comment by gpotter2** [comment]
What are you using `ObjectPipe` for exactly ? I should probably document more extensively it's behavior

**Comment by polybassa** [comment]
I've tried to use it instead of a `prioritized queue` and hoped, the select mechanisms safes some processing time. But unfortunately this was not the case.

**Comment by polybassa** [comment]
@gpotter2 Ready for merge

**Pull request merged**